submission/proxy_server.go

```diff
@@ -44,7 +45,10 @@ func NewProxyServer(logListPath string, dBuilder DistributorBuilder, reqTimeout
 // Run starts regular Log list updates. Blocks until initialization happens.
 func (s *ProxyServer) Run(logListRefreshInterval time.Duration, rootsRefreshInterval time.Duration) {
 	s.p.Run(context.Background(), logListRefreshInterval, rootsRefreshInterval)
-	<-s.p.Init
+	_, ok := <-s.p.Init
+	if !ok {
+		glog.Fatalf("unable to initialize Proxy")
```

Comment:
I think the only way this happens is if something else has already read the value from `s.p.Init`, since you only close the channel on success (and I believe that's the only scenario in which `ok == false`). Therefore, saying you were unable to initialize Proxy is false - it would be initialized.

Reply:
Also, this change is seemingly unrelated to the PR, since this code isn't exercised by the test you're changing.

Reply:
Changed back